scripts/components/networking/statuseffectcomponent/statuseffectcomponent.gd

```diff
@@ -0,0 +1,312 @@
+	if stack_override >= 0:
+		currStacks = stack_override
+		newStatus.default_stacks = stack_override
+	else:
+		currStacks = newStatus.default_stacks
+
+	# If another of this class exists, combine it.
+	if status_effects_active.has(newStatus.status_class):
+		assert(
+			status_effects_active[status_class].get("resource", false),
+			"There's no resource set for this status effect"
+		)
+
+		# If it can be stacked, add both togheter, otherwise, set duration to to highest of the two
+		if newStatus.combine_stackable_duration:
+			set_duration(status_class, currDuration)
+		else:
+			set_duration(status_class, max(get_duration(status_class), currDuration))
+
+		if newStatus.combine_stack_override:
+			set_stacks(status_class, max(get_stacks(status_class), currStacks))
+		else:
+			set_stacks(status_class, get_stacks(status_class) + currStacks)
+	else:
+		status_effects_active[status_class] = {}
+		set_duration(status_class, currDuration)
+		set_stacks(status_class, currStacks)
+		set_resource(status_class, newStatus)
+		set_applier(status_class, applier.get_name())
+		assert(get_resource(status_class) is StatusEffectResource)
+
+	get_resource(status_class).effect_applied(user, get_status_effect_data(status_class))
+	if Global.debug_mode:
+		GodotLogger.info(
+			"Added status effect '{0}' with {1} stacks and {2} duration.".format(
+				[status_class, get_stacks(status_class), get_duration(status_class)]
+			)
+		)
+
+
+## Server only
+func remove_status(status_class: String):
+	get_resource(status_class).effect_removed(user, get_status_effect_data(status_class))
+	status_effects_active.erase(status_class)
+
+
+func clear_status():
+	for status: String in status_effects_active:
+		remove_status(status)
+
+
+func set_duration(status_class: String, dur: float):
+	if not status_effects_active.has(status_class):
+		GodotLogger.error("This status is not present in this component.")
+		return
+
+	status_effects_active[status_class]["duration"] = dur
+
+
+func get_duration(status_class: String) -> float:
+	# Equivalent of return status_effects_active[status_class]["duration"]
+	# But safe.
+	return status_effects_active.get(status_class, {}).get("duration", 0.0)
+
+
+## If [param trigger_effect] is true, _effect_stack_changed() is called on the status.
+func set_stacks(status_class: String, sta: int, trigger_effect: bool = false):
+	if not status_effects_active.has(status_class):
+		GodotLogger.error("This status is not present in this component.")
+		return
+
+	status_effects_active[status_class]["stacks"] = sta
+	if trigger_effect:
+		get_resource(status_class)._effect_stack_change(user, get_status_effect_data(status_class))
+
+
+func get_stacks(status_class: String) -> int:
+	return status_effects_active.get(status_class, {}).get("stacks", 0)
+
+
+func set_resource(status_class: String, resource: StatusEffectResource):
+	if not status_effects_active.has(status_class):
+		GodotLogger.error("This status is not present in this component.")
```

Comment:
Is this really an error? Or should this loglevel be lowered to info?

Reply:
There's no situation where set_resource should be called for a status that's not applied.  
Since there would be nowhere to place the resource in the first place.  
I considered making it an assert but this seems a bit less "static" so it is more likely to break